Build failure: mysql-workbench

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Try to build mysql-workbench

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/301007541

### Relevant log output

```console
error: builder for '/nix/store/hb19c2ndkc80qd1bbnhq4fcz1sh8pv72-mysql-workbench-8.0.42.drv' failed with exit code 1;
       last 25 log lines:
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1816:37: note: remove the '< >'
       > [247/788] Building CXX object frontend/linux/linux_utilities/CMakeFiles/linux_utilities.dir/listmodel_wrapper.cpp.o
       > In file included from /build/mysql-workbench-community-8.0.42-src/backend/wbpublic/grt/tree_model.h:33,
       >                  from /build/mysql-workbench-community-8.0.42-src/frontend/linux/linux_utilities/listmodel_wrapper.h:46,
       >                  from /build/mysql-workbench-community-8.0.42-src/frontend/linux/linux_utilities/listmodel_wrapper.cpp:25:
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1388:40: warning: template-id not allowed for constructor in C++20 [-Wtemplate-id-cdtor]
       >  1388 |     explicit ListRef<internal::Integer>(const ValueRef &lvalue) : BaseListRef(lvalue) {
       >       |                                        ^
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1388:40: note: remove the '< >'
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1816:37: warning: template-id not allowed for constructor in C++20 [-Wtemplate-id-cdtor]
       >  1816 |     explicit ListRef<internal::Dict>(const ValueRef &lvalue) : BaseListRef(lvalue) {
       >       |                                     ^
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1816:37: note: remove the '< >'
       > [248/788] Building CXX object frontend/linux/linux_utilities/CMakeFiles/linux_utilities.dir/plugin_editor_base.cpp.o
       > In file included from /build/mysql-workbench-community-8.0.42-src/frontend/linux/linux_utilities/plugin_editor_base.h:34,
       >                  from /build/mysql-workbench-community-8.0.42-src/frontend/linux/linux_utilities/plugin_editor_base.cpp:25:
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1388:40: warning: template-id not allowed for constructor in C++20 [-Wtemplate-id-cdtor]
       >  1388 |     explicit ListRef<internal::Integer>(const ValueRef &lvalue) : BaseListRef(lvalue) {
       >       |                                        ^
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1388:40: note: remove the '< >'
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1816:37: warning: template-id not allowed for constructor in C++20 [-Wtemplate-id-cdtor]
       >  1816 |     explicit ListRef<internal::Dict>(const ValueRef &lvalue) : BaseListRef(lvalue) {
       >       |                                     ^
       > /build/mysql-workbench-community-8.0.42-src/library/grt/src/grt.h:1816:37: note: remove the '< >'
       > ninja: build stopped: subcommand failed.
       For full logs, run:
         nix log /nix/store/hb19c2ndkc80qd1bbnhq4fcz1sh8pv72-mysql-workbench-8.0.42.drv
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.2, NixOS, 25.11 (Xantusia), 25.11.20250617.9e83b64`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/ny8c07vsrfwcb1c4i3jcpbi3qi4w9wy6-source`


### Notify maintainers


@TomaSajt 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/my/mysql-workbench/package.nix
+++ b/pkgs/by-name/my/mysql-workbench/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchurl,
+  fetchpatch,
   replaceVars,
 
   cmake,
@@ -73,6 +74,13 @@ stdenv.mkDerivation (finalAttrs: {
 
     # Don't try to override the ANTLR_JAR_PATH specified in cmakeFlags
     ./dont-search-for-antlr-jar.patch
+
+    # fixes the build with python 3.13
+    (fetchpatch {
+      name = "python3.13.patch";
+      url = "https://git.pld-linux.org/?p=packages/mysql-workbench.git;a=blob_plain;f=python-3.13.patch;h=d1425a93c41fb421603cda6edbb0514389cdc6a8;hb=bb09cb858f3b9c28df699d3b98530a6c590b5b7a";
+      hash = "sha256-hLfPqZSNf3ls2WThF1SBRjV33zTUymfgDmdZVpgO22Q=";
+    })
   ];
 
   postPatch = ''

